src/components/HdModal.vue

```diff
@@ -1,139 +1,91 @@
 <template>
   <transition name="modal">
-    <section
-      ref="modal"
-      :class="{
-        'modal--no-spacing': noSpacing,
-        'modal--no-external-spacing': noExternalSpacing,
-        'modal--no-close-icon': !showCloseIcon,
-        'modal--external-close-icon': externalCloseIcon,
-        [`modal--mobile-align-${mobileAlign}`]: true,
-        [`modal--spacing-${spacingSize}`]: true,
-      }"
-      class="modal"
-      @click="onCloseClick"
+    <article
+      ref="hd-modal"
+      class="hd-modal"
+      :class="{ 'hd-modal--with-icon': withIcon }"
     >
-      <div
-        :class="`modal__overlay--${overlayColor}`"
-        class="modal__overlay"
-      />
-      <div
-        :style="{
-          width: width > 0 ? `${width}px` : null,
-        }"
-        class="modal__inner"
-        @click.stop
-      >
-        <header
-          v-if="$slots.header"
-          class="modal__header"
-        >
-          <slot name="header" />
+      <div class="hd-modal__overlay" />
+      <div class="hd-modal__container">
+        <header class="hd-modal__header">
+          <hd-icon
+            v-if="withIcon"
+            :src="iconSrc"
+            class="hd-modal__modal-icon"
+          />
+          <h3 class="hd-modal__title">
+            <slot name="title" />
+          </h3>
         </header>
 
-        <div
-          v-if="$slots.content"
-          class="modal__body"
-        >
-          <slot name="content" />
+        <div class="hd-modal__body">
+          <slot name="body" />
         </div>
 
-        <div
-          v-if="actions.length > 0"
-          class="modal__actions"
+        <footer
+          class="hd-modal__footer"
+          :class="{ 'hd-modal__footer--is-wide': isWide }"
         >
-          <button
-            v-for="(action, i) in actions"
-            :key="i"
-            :disabled="action.disabled"
-            :class="{
-              'btn--primary': action.type === 'primary',
-              'btn--ghost': action.type !== 'primary',
-            }"
-            class="modal__actions__action btn"
-            type="button"
-            @click="action.callback"
-          >
-            {{ action.label }}
-          </button>
-        </div>
+          <slot name="actions">
+            <hd-button
+              modifier="primary"
+              @click="$emit('close')"
+            >
+              OK
```

Comment:
Should we update this part?

Reply:
It's a default action. Most times it won't be on the component. And "OK" is a common word in German and English.

I don't see a reason to remove it at the moment.

Reply:
https://github.com/homeday-de/homeday-blocks/pull/732/commits/14fb11a56c6e06765c1b850f45180b5e03f87a98